detector: fix an issue with deleting all teams/tags/authorized users/teas=ms

dwalasek opened the pull request:
detector: fix an issue with deleting all teams/tags/authorized users/authorized teams

To remove teams/tags/authorized users/authorize from detector resource, the API client must send an empty array. Sending any of these fields missing (null) has no effect

dwalasek committed:
detector: fix an issue with deleting all teams/tags/authorized users/authorized teams from detector object

To remove teams/tags/authorized users/authorize from detector resource, the API client must send an empty array. Sending any of these fields missing (null) has no effect

atoulme commented on go.mod:
```diff
@@ -8,7 +8,7 @@ require (
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.30.0
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/signalfx/signalfx-go v1.36.0
+	github.com/signalfx/signalfx-go v1.36.1-0.20240613150536-43123c77db5e
```
Why not use 1.37.0?

dwalasek replied:
Fixed it.

atoulme commented on signalfx/resource_signalfx_detector.go:
```diff
@@ -306,7 +306,7 @@ func getPayloadDetector(d *schema.ResourceData) (*detector.CreateUpdateDetectorR
 	minDelay := int32(d.Get("min_delay").(int) * 1000)
 
-	var tags []string
+	tags := []string{}
```
Any chance you can add a test for this?

dwalasek replied:
Good point. I've added some scenarios to: resource_signalfx_detector_test

dwalasek committed:
detector: fix an issue with deleting all teams/tags/authorized users/authorized teams from detector object

To remove teams/tags/authorized users/authorize from detector resource, the API client must send an empty array. Sending any of these fields missing (null) has no effect
